scripts/india_census/primary_religion_data/preprocess.py

```diff
@@ -137,92 +212,135 @@ def stat_var_index_key(row, data_category_column=None):
         # Export it as CSV. It will have the following columns
         # Name,TRU,columnName,value,StatisticalVariable,Year
         if path.exists(self.csv_file_path):
+            self.raw_df.drop_duplicates(
+                subset=["census_location_id", "StatisticalVariable", "Year"],
+                keep="first",
+                inplace=True)
             self.raw_df.to_csv(self.csv_file_path,
+                               quoting=csv.QUOTE_ALL,
                                mode='a',
                                index=False,
                                header=False)
         else:
-            self.raw_df.to_csv(self.csv_file_path, index=False, header=True)
+            self.raw_df.drop_duplicates(
+                subset=["census_location_id", "StatisticalVariable", "Year"],
+                keep="first",
+                inplace=True)
+            self.raw_df.to_csv(self.csv_file_path,
+                               quoting=csv.QUOTE_ALL,
+                               index=False,
+                               header=True)
 
     def _get_base_name(self, row):
         # To make the name meaningful add Religion to the
         # the name of the stat var
         name = "Count_" + row["populationType"] + "_Religion"
         return name
 
+    def _get_base_constraints(self, row):
+        constraints = ""
+        return constraints
+
     def _get_stat_var_name(self, name):
         return "dcid:indianCensus/{}".format(name)
 
     def _get_measured_property_name(self, name):
-        return "dcs:indianCensus/{}".format(name)
+        return "count"
 
     def _create_variable(self,
                          data_row,
                          place_of_residence=None,
                          data_category=None):
         row = copy.deepcopy(data_row)
         name_array = []
+        constraints_array = []
 
         name_array.append(self._get_base_name(row))
 
-        if data_category:
-            name_array.append(title_case(data_category))
-            row["description"] = row["description"] + " - " + data_category
+        # No need to add empty constraint to the list
+        if self._get_base_constraints(row) != "":
+            constraints_array.append(self._get_base_constraints(row))
+
+        name_array.append(title_case(data_category))
+        row["description"] = row["description"] + " - " + data_category
+        constraints_array.append("religion: " +
+                                 _RELIGION_DCID_MAPPING[data_category])
 
         if row["age"] == "YearsUpto6":
             name_array.append("YearsUpto6")
+            constraints_array.append("age: dcid:YearsUpto6")
```

Comment:
Quantity ranges can also be specified as "age: [- 6 Years]".

It will be internally converted into dcid:YearsUpto6